test/Microsoft.SqlTools.ServiceLayer.IntegrationTests/ObjectManagement/ObjectManagementServiceTests.cs

```diff
@@ -0,0 +1,150 @@
+//
+using System;
+using System.Collections.Concurrent;
+using System.Threading.Tasks;
+using Microsoft.SqlServer.Management.Smo;
+using Microsoft.SqlTools.Hosting.Protocol;
+using Microsoft.SqlTools.ServiceLayer.Connection;
+using Microsoft.SqlTools.ServiceLayer.IntegrationTests.Utility;
+using Microsoft.SqlTools.ServiceLayer.ObjectManagement;
+using Microsoft.SqlTools.ServiceLayer.ObjectManagement.Contracts;
+using Microsoft.SqlTools.ServiceLayer.QueryExecution;
+using Microsoft.SqlTools.ServiceLayer.QueryExecution.DataStorage;
+using Microsoft.SqlTools.ServiceLayer.SqlContext;
+using Microsoft.SqlTools.ServiceLayer.Test.Common;
+using Moq;
+using NUnit.Framework;
+using static Microsoft.SqlTools.ServiceLayer.IntegrationTests.Utility.LiveConnectionHelper;
+
+namespace Microsoft.SqlTools.ServiceLayer.IntegrationTests.ObjectManagement
+{
+    public class ObjectManagementServiceTests
+    {
+        private const string TableQuery = @"CREATE TABLE testTable1_RenamingTable (c1 int)";
+        private const string OWNERURI = "testDB";
+        private ObjectManagementService objectManagementService;
+        private SqlTestDb testDb;
+        private Mock<RequestContext<bool>> requestContextMock;
+
+        [SetUp]
+        public async Task TestInitialize()
+        {
+            this.testDb = await SqlTestDb.CreateNewAsync(serverType: TestServerType.OnPrem, query: TableQuery, dbNamePrefix: "RenameTest");
+
+            requestContextMock = new Mock<RequestContext<bool>>();
+            ConnectionService connectionService = LiveConnectionHelper.GetLiveTestConnectionService();
+
+            TestConnectionResult connectionResult = await LiveConnectionHelper.InitLiveConnectionInfoAsync(testDb.DatabaseName, OWNERURI, ConnectionType.Default);
+
+            ObjectManagementService.ConnectionServiceInstance = connectionService;
+            this.objectManagementService = new ObjectManagementService();
+        }
+
+        [TearDown]
+        public async Task TearDownTestDatabase()
+        {
+            await SqlTestDb.DropDatabase(testDb.DatabaseName);
+        }
+
+        [Test]
+        public async Task TestRenameTable()
+        {
+            //arrange & act
+            await objectManagementService.HandleRenameRequest(this.InitRequestParams("RenamingTable", String.Format("Server/Database[@Name='{0}']/Table[@Name='testTable1_RenamingTable' and @Schema='dbo']", testDb.DatabaseName)), requestContextMock.Object);
+
+            //assert
+            requestContextMock.Verify(x => x.SendResult(It.Is<bool>(r => r == true)));
+
+            Query queryRenameObject = ExecuteQuery("SELECT * FROM " + testDb.DatabaseName + ".sys.tables WHERE name='RenamingTable'");
+            Assert.IsTrue(queryRenameObject.HasExecuted);
+            Assert.IsFalse(queryRenameObject.HasErrored);
+            Assert.AreEqual(1, queryRenameObject.Batches[0].ResultSets[0].RowCount);
```

Comment:
For these asserts it's preferred to use the fluent (constraint) style assertions (https://docs.nunit.org/articles/nunit/writing-tests/assertions/assertion-models/constraint.html) - e.g. this would be 

`Assert.That(queryRenameObject.Batches[0].ResultSets[0].RowCount, Is.EqualTo(1))`

They're a lot easier to read, are harder to mess up order of expected vs actual and have better error messages when they do fail. (all assertions in these tests can be changed to use `Assert.That(...)`)

In addition when you do this make sure you add a message to each of them that indicates what failed, since otherwise you'll usually just end up with a message like "0 does not equal 1" which isn't helpful, especially when there's multiple similar assertions in the same test. 

e.g. 

`Assert.That(queryRenameObject.Batches[0].ResultSets[0].RowCount, Is.EqualTo(1), "Did not find table with new name after rename")`

Reply:
Yeah sure, I can do that. Thank you very much for the tipp. Are you ok, with the error messages (not a native english speaker! 😉)?

```c#
 Query queryRenameObject = ExecuteQuery("SELECT * FROM " + testDb.DatabaseName + ".sys.tables WHERE name='RenamingTable'");
            Assert.That(queryRenameObject.HasExecuted, Is.True, "The query to check for the renamed table was not executed");
            Assert.That(queryRenameObject.HasErrored, Is.False, "There were errors on the execution of the query to check for the renamed table");
            Assert.That(queryRenameObject.Batches[0].ResultSets[0].RowCount, Is.EqualTo(1), "Did not find the table with the new name after the rename operation");

            Query queryOldObject = ExecuteQuery("SELECT * FROM " + testDb.DatabaseName + ".sys.tables WHERE name='testTable1_RenamingTable'");
            Assert.That(queryOldObject.HasExecuted, Is.True, "The query to check for the old table was not executed");
            Assert.That(queryOldObject.HasErrored, Is.False, "There were errors on the execution of the query to check for the old table");
            Assert.That(queryOldObject.Batches[0].ResultSets[0].RowCount, Is.EqualTo(0), "Did find the old table which should have been renamed");
        
```

Reply:
Yeah, those are fine! It's mostly just to have a unique phrase so it's clear where exactly the failure happened. 

(and do this for the rest of the tests in this file too)